src/lib/GUID.cpp

```diff
@@ -70,20 +122,72 @@ bool GUID::is_equal(GUID other)
 }
 
 // This is used for FileDataStores. The GUID is given as a string in the file
-void GUID::from_string(std::string str)
+void GUID::from_string(std::string const str)
```

Comment:
I actually wonder, seeing that you've made all these other constructors, might it be possible to rename this to `GUID::GUID(std::string const str)` while at it? :)

Reply:
Sure, i'll make it a constructor. I needed these to conveniently prepare tests. This is the only place where i needed it.

I guess this could be made a debug only feature.

Reply:
Maybe making the extra constructors a test-only feature would be smarter.
Well for now, i just added a constructor which calls `from_string()`. Or do you want to get rid of it?

Reply:
I'd rather get rid of it because we don't have any users of `from_string()` at this point, so might as well. If that's not too much of a bother!

Let's leave the other constructors for now, as the created objects' contents are actually tested, so I guess that's proof enough that they work.

Reply:
Ok, this feature is removed =)

Reply:
Ohhh! No, sorry, I was just talking about moving method itself to a `GUID::GUID`(string) constructor, i.e. renaming it. I wasn't planning on nuking all this work you did!

Reply:
Oh, haha, i didn't mind. I'll reset it.
In this context i noticed, i missed the most important test, the test for parsing.